diag_manager/include/fms_diag_fieldbuff_update.fh

```diff
@@ -0,0 +1,1336 @@
+    ENDIF
+
+    succeded = FMS_DIAG_FBU_PNAME_ (ofield_cfg, ofield_index_cfg, field_ptr, sample, &
+      & ofb_ptr, ofc_ptr, bbounds, count_0d, num_elements, mask_ptr, weight1, missvalue,  &
+      & field_num_threads, field_active_omp_level, issued_mask_ignore_warning, &
+      & l_start, l_end, err_msg,  err_msg_local)
+  END FUNCTION FMS_DIAG_FBU_3D_PNAME_
+
+
+!> @brief This code will be used by the preprecessor to generate an implementation
+!! to the module procudure for the fieldbuff_update interface.
+!! Updates elements of the running field output buffer (argument ofb)
+!! and counter (argument ofc) based on the input field data array (argument field_d).
+!! In general the formulas are :
+!! A) ofb(l) = ofb(l) + (weight * field(l))**pow_value
+!! B) ofc(l) = ofc(l) + weight
+!! where l is a standing for some set of indecies in multiple dimensions.
+!! Note this function may set field object members active_omp_level and  num_threads.
+!> @addtogroup fms_diag_fieldbuff_update_mod
+!> @{
+   FUNCTION FMS_DIAG_FBU_PNAME_ (ofield_cfg, ofield_index_cfg, field_d, sample, &
+    & ofb, ofc, bbounds, count_0d, num_elements, mask, weight1, missvalue,  &
+    & field_num_threads, field_active_omp_level, issued_mask_ignore_warning, &
+    & l_start, l_end, err_msg,  err_msg_local ) result( succeded )
+      TYPE(fms_diag_outfield_type), INTENT(in) :: ofield_cfg  !< The ofield_cfg object
+      TYPE(fms_diag_outfield_index_type) , INTENT(in) :: ofield_index_cfg  !< The ofield_index_cfg object
+      FMS_DIAG_FBU_DATA_TYPE_ , CONTIGUOUS, DIMENSION(:,:,:,:), INTENT(in) :: field_d !< The input field data array.
+      FMS_DIAG_FBU_DATA_TYPE_ , DIMENSION(:,:,:,:,:), INTENT(inout) :: ofb !< Output Field Buffer
+      FMS_DIAG_FBU_DATA_TYPE_ , DIMENSION(:,:,:,:,:), INTENT(inout) :: ofc !< Output Field Counter
+      TYPE(fms_diag_ibounds_type), INTENT(inout) :: bbounds !< An instance of the class fms_diag_buff_intervals_t
+      INTEGER, INTENT(in) :: sample  !< The index along the diurnal time axis
+      FMS_DIAG_FBU_DATA_TYPE_, INTENT(inout):: count_0d !< Normally the member of the buffer object of the same name.
+      INTEGER, INTENT(inout) :: num_elements
+
+      LOGICAL, CONTIGUOUS, DIMENSION(:,:,:,:), INTENT(in), OPTIONAL :: mask !< mask
+      FMS_DIAG_FBU_DATA_TYPE_, INTENT(in) :: weight1 !< Field data is multiplied by weight
+      FMS_DIAG_FBU_DATA_TYPE_, INTENT(in) :: missvalue !< Buffer may be set to missvalue where mask is false.
+
+      INTEGER, INTENT(inout) :: field_num_threads
+      INTEGER, INTENT(inout) :: field_active_omp_level
+
+      LOGICAL, INTENT(inout) :: issued_mask_ignore_warning
+      INTEGER, DIMENSION(3), INTENT(in)  :: l_start !< local start indices on 3 axes for regional output
+      INTEGER, DIMENSION(3), INTENT(in)  :: l_end !< local end indices on 3 axes for regional output
+      CHARACTER(len=*), INTENT(inout), OPTIONAL :: err_msg
+      CHARACTER(len=256), INTENT(inout) :: err_msg_local
```

Comment:
These need to be documented.  Why do you have err_msg and err_msg_local both coming in here?  Shouldn''t err_msg_local only be used for a specific subroutine, and only used is `err_msg` in included.

Reply:
They may both be passed to the error handler, plus the error_message_local may set in the bounds checking function.
Added some documentation,